tooling/minimize/intrinsics/src/lib.rs

```diff
@@ -29,3 +33,96 @@ pub unsafe fn deallocate(ptr: *mut u8, size: usize, align: usize) {
     let layout = Layout::from_size_align(size, align).unwrap();
     unsafe { System.deallocate(ptr, layout); }
 }
+
+
+static JOIN_HANDLES: Mutex<Vec<Option<JoinHandle<()>>>> = Mutex::new( Vec::new() );
+
+struct SendPtr<T>(*const T);
+unsafe impl<T> Send for SendPtr<T> {}
+
+pub fn spawn(fn_ptr: extern "C" fn(*const ()), data_ptr: *const ()) -> usize {
+    let mut join_handles = JOIN_HANDLES.lock().unwrap();
+
+    if join_handles.len() == 0 {
+        join_handles.push(None);
```

Comment:
What is the point of this?

Reply:
In the last version there was a weird off by one. This could also be implemented using Lazy but I was unsure if we should add dependencies on such crates in the intrinsics.

Reply:
Off by one in which sense?

The IDs returned by the intrinsics should be treated as opaque, so it doesn't matter whether this and MiniRust return the same IDs.